### ISSUE TITLE
Redshift - max_index_keys Error

### ISSUE THREAD

**Issue by traytonwhite** [yesodweb/persistent#713]
Hi there - when trying to connect to Redshift, I seem to be able to connect just fine - but I'm running into this error: 

```*** Exception: SqlError {sqlState = "42501", sqlExecStatus = FatalError, sqlErrorMsg = "must be superuser to examine \"max_index_keys\"", sqlErrorDetail = "", sqlErrorHint = ""}```

I've been Googling around, and it seems related to the connector library trying some very Postgres-specific items that AWS has disabled for Redshift.

I'm happy to troubleshoot further, but I'm at a loss on how to track down where the query for `max_index_keys` occurs. I think I need to know: is it at the persistent layer, postgresql-simple, or even further down within my libpq library?

I'm hoping for either guidance on how to further troubleshoot this, or suggestions on ways to workaround this Redshift-specific limitation that others have had success with.

Thank you!

**Comment by MaxGabriel** [comment]
So I found this AWS thread about the issue: https://forums.aws.amazon.com/thread.jspa?threadID=133514

It appears to be triggered by querying certain things from the `information_schema` database in postgres. Notably modejosh in that thread said this will trigger it:

```
SELECT COUNT(1) FROM information_schema.key_column_usage
SELECT COUNT(1) FROM information_schema.constraint_column_usage
```

I searched the Persistent codebase and found all `information_schema` queries happen in https://github.com/yesodweb/persistent/blob/master/persistent-postgresql/Database/Persist/Postgresql.hs. 

The one that matches up best to the AWS reports is this one: https://github.com/yesodweb/persistent/blob/ed17dc390767337108b99b377a204914e591297c/persistent-postgresql/Database/Persist/Postgresql.hs#L659-L673

Here's what I would do to troubleshoot: Open `psql` for your Redshift Cluster and try the various information_schema queries that are used in `Postgresql.hs`, and see which ones fail. Possibly these queries can be rewritten such that they don't fail (I'm not sure if that's possible, depending on much Redshift locks down this information).

I also noticed that the `key_column_usage` query is in `getColumns`, which is only used by `migrate'`. Possibly if you don't try to have Persistent migrate your Redshift database, it will avoid this problem altogether?
